### ISSUE TITLE
Instance variables in subclasses sometimes are defined incorrectly

### ISSUE THREAD

**Issue by wagenet** [ruby/typeprof#92]
In my specific case, it is overwriting the parent type with something overly specific:

```rbs
class Base
  @warnings_suppressed: bool
end

class Subclass < Base
  @warnings_suppressed: bot
end
```

Rough code:

```rb
class Base
  def initialize
    @warnings_suppressed = false
  end

  def suppress_warnings!
    tap { @warnings_suppressed = true }
  end

  def warn(message, event = nil)
    return if @warnings_suppressed
    puts message
  end
end

class Subclass < Base
  def do_stuff
    warn("Something went wrong")
  end
end
```

**Comment by mame** [comment]
Thank you! Is this fixed by your pull request #94?

**Comment by wagenet** [comment]
Yes, this should be fixed now.
